source/Ship.cpp

```diff
@@ -64,31 +64,32 @@ namespace {
 
 	const double MAXIMUM_TEMPERATURE = 100.;
 
-	// Scanning takes between 2 and 10 seconds (SCAN_TIME / MAX_SCAN_STEPS and SCAN_TIME / MIN_SCAN_STEPS)
+	// Scanning takes up to 10 seconds (SCAN_TIME / MIN_SCAN_STEPS)
 	// dependent on the range from the ship (among other factors).
 	// The scan speed uses a gaussian drop-off with the reported scan radius as the standard deviation.
 	const double SCAN_TIME = 600.;
 	// Always gain at least 1 step of scan progress for every frame spent scanning while in range.
 	// This ensures every scan completes within 600 frames (10 seconds) of being in range.
 	const double MIN_SCAN_STEPS = 1;
-	// Gain no more than 5 steps of scan progress for every frame spent scanning while in range.
-	// This ensures no scan takes fewer than 120 frames (2 seconds.)
-	const double MAX_SCAN_STEPS = 5; // minimum of 2 seconds to scan
+	// Exponential dropoff of scan rate starts at scan rate of SCAN_TIME/LINEAR_RATE
+	// with an exponent of SCAN_DROPOFF_EXPONENT.
+	const double LINEAR_RATE = 5;
+	const double SCAN_DROPOFF_EXPONENT = 5;
 
 	// These numbers ensure it takes 10 seconds for a Cargo Scanner to scan
 	// a Bulk Freighter at point blank range. Any ship with less than 40
 	// cargo space takes as long as a ship with 40 cargo space.
 	const double SCAN_MIN_CARGO_SPACE = 40;
-	const double SCAN_CARGO_FACTOR = 3;
+	const double SCAN_CARGO_FACTOR = pow(1.0 / sqrt(5), -1.5) / 600;
 
 	// This ensures it takes 10 seconds for an Outfit Scanner to scan a
 	// Bactrian at point blank range. Any ship with less than 200 outfit
 	// space takes as long as a ship with 200 outfit space.
 	const double SCAN_MIN_OUTFIT_SPACE = 200;
-	const double SCAN_OUTFIT_FACTOR = 10;
+	const double SCAN_OUTFIT_FACTOR = pow(1.0 / sqrt(15), -1.5) / 740;
 
 	// Formula for the scan outfit or cargo factor is:
-	// factor = pow(sqrt(scanEfficiency) * framesToFullScan / SCAN_TIME, 1.5) / referenceSize
+	// factor = pow(sqrt(scanEfficiency) * framesToFullScan / SCAN_TIME, -1.5) / referenceSize
```

Comment:
comments usually appear first, and not last. I'd say move this up

Reply:
I rearranged the comments. Does it make more sense now?